api/controllers/StatsController.js

```diff
@@ -96,20 +102,76 @@ function transformSeasonToDTO(season) {
   return {
     ...rest,
     rankings: rankingsAsArray,
+    uniquePlayersPerWeek: season.uniquePlayersPerWeek.map((playerSet) => playerSet.size),
   };
 }
 
+/**
+ * Returns copy of seasons with the initialized arrays for
+ * gameCounts and uniquePlayersPerWeek
+ * @param {
+*  name: string
+*  startTime: int
+*  endTime: int
+*  firstPlace: int (playerId)
+*  secondPlace: int (playerId)
+*  thirdPlace: int (playerId)
+*  fourthPlace: int (playerId)
+*  rankings: Map<int: playerId, {
+*    username: string
+*    matches: Map<int: weekNum, Array<{opponent: string, result: Result}>>
+*  }>
+* } seasons
+* 
+* @returns {
+*  name: string
+*  startTime: int
+*  endTime: int
+*  firstPlace: int (playerId)
+*  secondPlace: int (playerId)
+*  thirdPlace: int (playerId)
+*  fourthPlace: int (playerId)
+*  rankings: Map<int: playerId, {
+*    username: string
+*    matches: Map<int: weekNum, Array<{opponent: string, result: Result}>>
+*  }>
+*  gameCounts: Array<int>
+*  uniquePlayersPerWeek: Array<Set<int: playerIds>>
```

Comment:
The `uniquePlayerCountsPerWeek` uses a `Set()` for each week to de-dupe the players by their id, then converts to the size of that set for the response sent back clientside. This makes for relatively efficient processing without sending usernames or ids to the client beyond the ones used for the ranked leaderboard